iocore/net/quic/QUICStreamState.cc

```diff
@@ -292,15 +287,26 @@ QUICSendStreamState::is_allowed_to_receive(const QUICFrame &frame) const
 bool
 QUICSendStreamState::is_allowed_to_receive(QUICFrameType type) const
 {
+  // An endpoint that receives a RESET_STREAM frame for a send-only stream MUST terminate the connection with error
+  // PROTOCOL_VIOLATION. 
+  // PS: RST_STREAM should be processed before Stream transition.
```

Comment:
I don't think we should assume this. We cannot send PROTOCOL_VIOLATION (from the caller) if we assert here.